gui/default/syncthing/settings/settingsModalView.html

```diff
@@ -159,6 +159,197 @@
+                    <th>
+                      <label>
+                        <span translate="settings.webauthn.credentials.columnHeading.require2fa">2FA</span>
+                        <a href="{{docsURL('users/webauthn#webauthn-require2fa')}}">
+                          <span class="fa fa-question-circle" title="Open help: 2FA setting for WebAuthn credentials" />
+                        </a>
+                      </label>
+                    </th>
+                    <th colspan="1000"><label translate>Actions</label></th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <tr ng-repeat="cred in webauthn.state.credentials | filter:{rpId: tmpGUI.webauthnRpId}:true">
+                    <td ng-if="!webauthn.editingCredentialIds[cred.id]" class="no-overflow-ellipse stretch">
+                      {{ cred.nickname || (cred.id | limitTo: 20) }}
+                    </td>
+                    <td ng-if="webauthn.editingCredentialIds[cred.id]" class="no-overflow-ellipse stretch">
+                      <input class="form-control" type="text" ng-model="cred.nickname" placeholder="{{ cred.id }}" />
+                    </td>
+
+                    <td class="small-data" title="{{cred.createTime | date:'yyyy-MM-dd HH:mm:ss'}}">
+                      {{cred.createTime | date:'yyyy-MM-dd'}}
+                    </td>
+                    <td class="small-data" title="{{ cred.lastUseTime | date:'yyyy-MM-dd HH:mm:ss' }}">
+                      {{ cred.lastUseTime | date:'yyyy-MM-dd' }}
+                    </td>
+
+                    <td class="small-data text-center">
+                      <input type="checkbox" ng-model="cred.requireUv" />
+                    </td>
+                    <td class="small-data button-cell">
+                      <button type="button" class="btn btn-link btn-sm" ng-click="webauthn.editingCredentialIds[cred.id] = true" ng-disabled="webauthn.editingCredentialIds[cred.id]">
+                        <span class="fas fa-edit"></span>&nbsp;<span translate>Rename</span>
+                      </button>
+                    </td>
+                    <td class="small-data button-cell">
+                      <button type="button" class="btn btn-link btn-sm" ng-click="deleteWebauthnCredential(cred)">
+                        <span class="fas fa-trash"></span>&nbsp;<span translate>Delete</span>
+                      </button>
+                    </td>
+                  </tr>
+
+                  <tr>
+                    <td colspan="4">
+                      <p class="text-info" ng-if="webauthnAvailable() && !config.gui.user" translate>
+                        Set a username and save to enable WebAuthn.
+                      </p>
+
+                      <p class="text-info" ng-if="!tmpGUI.useTLS" translate>
+                        WebAuthn requires HTTPS.
```

Comment:
This check is not really accurate, as HTTPS is always available on the Syncthing GUI port.  The option just causes the backend to force a redirect to HTTPS.  If that is not set, and someone tries to log in via HTTP, they will get the error message at that time.  I guess we can drop the check here completely.

Reply:
Ok. Yeah, I'll just remove it. I recently also realized that `localhost` is a special case that is considered a "secure context" even on naked HTTP, so for many users this warning is even a false positive.

Reply:
Oh, it is useful on the login page for the *current* location.  I didn't mean you should remove it there.  Only the check about the `useTls` config option in the settings.

If you want to treat localhost specially, that should be excluded in `isLocationInsecure()`.

Reply:
Yeah, I dropped it due to the [problems with multiple origins](https://github.com/syncthing/syncthing/pull/9175#discussion_r1730310146) I mentioned earlier. But you're right; I reworked this along with the [changes to allow multiple origins](https://github.com/syncthing/syncthing/pull/9175#discussion_r1730327630). Now these checks look at the current origin instead of `guiCfg.useTLS`, and consider `localhost` a special case in `isLocationInsecure()`.

The checks on the settings page are still relevant, because you can easily log in to the settings GUI on `http://` if HTTPS is not currently forced, but you'll have to reload on `https://` in order to set up WebAuthn if the domain is not `localhost`.

Reply:
I think this is resolved now?